src/context/ImageResults.js

```diff
@@ -30,10 +34,12 @@ export const ImageResultsContextProvider = ({children}) => {
 
     const searchBarHandler = (inputVal) => imageListReq(inputVal)
 
-    const setSingleImageDataHandler = (image) => setSingleImageData(image)
+    const setSingleImageDataHandler = (image) => setSingleImageData(image) 
+
+    const handlePaginate = (paginateQuery, paginateNumber) => {imageListReq(paginateQuery, paginateNumber); setRandomPage(paginateNumber); console.log(paginateNumber)}
 
     return (
-    <ImageResultsContext.Provider value={{imageData, imagePreference: imagePreferenceHandler, inputImageSearch: searchBarHandler, singleImageData, setImageData: setSingleImageDataHandler}}>
+    <ImageResultsContext.Provider value={{imageData, imagePreference: imagePreferenceHandler, inputImageSearch: searchBarHandler, singleImageData, setImageData: setSingleImageDataHandler, query, paginate:handlePaginate}}>
```

Comment:
Can you create an object (e.g const imageResultsValues = {....}) to hold all those values that was assigned to the  Provider so it can become something like this
`<ImageResultsContext.Provider value={imageResultsValues}>{children}</ImageResultsContext.Provider>` ?

Reply:
Thanks, sure thing